### DISCUSSION TITLE
taillwind is not working when i use templete literals

### DISCUSSION THREAD

**Discussion by Neerajrawat123** [tailwindlabs/tailwindcss#12533] [Help]
tailwind is not changing styles although classes  has changed in browser  when i use templete literal to moniter the state changed by  button element . 

import React, { useEffect, useState } from "react";
import { cards } from "../constants/card.jsx";
import { IoIosArrowRoundForward } from "react-icons/io";
import { IoIosArrowRoundBack } from "react-icons/io";

function HousingEdgeSec() {
    const [rightClick, setRightClick] = useState(0);

  const handleRightButton = () => {
    setRightClick((prevRightClick) => prevRightClick + 1);
  };

  const handleLeftButton = () => {
    setRightClick((prevRightClick) => Math.max(prevRightClick - 1, 0));
  };

  const getclassNames = () => {
    let str =  'flex gap-3 pt-2 transform transition-transform ease-in-out right-0 ' ;
    if(rightClick>0){
      let ss = rightClick*160
      str+= `translate-x-${rightClick}`
    }
    return str
  };

  
   

    
  

  const Card = ({ name, url }) => {
    return (
      <div className="w-36 border h-36 pl-3 pt-3 rounded-lg hover:shadow-2xl bg-white ">
        <div className="h-16 w-16">
          <img src={url} alt="" />
        </div>
        <div className="mt-2 font-bold">
          <span>{name}</span>
        </div>
      </div>
    );
  };

  return (
    <div className="w-5/6  m-auto px-6 items-center bg-slate-100 mb-20 pb-14 pt-8">
    <div className="flex">
        <h1 className="text-4xl mr-4">Housing edge</h1>
        <button className="border border-red-600 text-red-600 px-4 py-2 rounded-md">
          Explore more
        </button>
      </div>

      <div className="mt-6  flex items-center overflow-hidden relative">
        {rightClick > 0 && (
          <button onClick={handleLeftButton} className=" absolute left-0 bg-white rounded-full z-10 p-2 shadow-lg">
            <IoIosArrowRoundBack  size={40} fontSize={1}/>
          </button>
        )}
        <div className={getclassNames()}>
          {cards.map((card, index) => (
            <Card key={index} name={card.name} url={card.url} />
          ))}
        </div>
        {rightClick < 5 && (
          <button onClick={handleRightButton} className="absolute right-0 bg-white rounded-full z-10 p-2 shadow-lg">
            <IoIosArrowRoundForward size={40} />
          </button>
        )}
      </div>
    </div>
  );
}

export default HousingEdgeSec;

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

I can't quite see some of the code since it does not seem to have been formatted correctly in the post, but you could consider using the `style` attribute instead.

**Comment by Zehad-10** [comment]
```
const getclassNames = () => {
  let str = 'flex gap-3 pt-2 transform transition-transform ease-in-out right-0 ';
  if (rightClick > 0) {
    let ss = rightClick * 160;
    str += `translate-x-${ss}`;
  }
  return str;
};`

Try this. Hopefully, it will work
you are defining the `ss` value but not using it
